diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index 94c713470..044f53554 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -1331,7 +1331,10 @@ UrlbarUtils.RESULT_PAYLOAD_SCHEMA = {
   [UrlbarShared.RESULT_TYPE.TAB_SWITCH]: {
     type: "object",
     required: ["url"],
     properties: {
+      zenEssentialTarget: {
+        type: "string",
+      },
       action: {
         type: "object",
         properties: {
