diff --git a/browser/components/urlbar/content/UrlbarInput.mjs b/browser/components/urlbar/content/UrlbarInput.mjs
index 95e433386..45785b36d 100644
--- a/browser/components/urlbar/content/UrlbarInput.mjs
+++ b/browser/components/urlbar/content/UrlbarInput.mjs
@@ -1727,5 +1727,6 @@ ${
         if (
-          this.hasAttribute("action-override") ||
-          (lazy.UrlbarPrefs.get("secondaryActions.switchToTab") &&
-            element?.dataset.action !== "tabswitch")
+          !result.payload.zenEssentialTarget &&
+          (this.hasAttribute("action-override") ||
+            (lazy.UrlbarPrefs.get("secondaryActions.switchToTab") &&
+              element?.dataset.action !== "tabswitch"))
         ) {
@@ -1760,12 +1761,23 @@ ${
-        let switched = this.window.switchToTabHavingURI(
-          Services.io.newURI(url),
-          true,
-          loadOpts,
-          lazy.UrlbarProviderOpenTabs.isNonPrivateUserContextId(
-            result.payload.userContextId
-          )
-            ? result.payload.userContextId
-            : null,
-          activeSplitView
-        );
+        const isZenEssentialResult = !!result.payload.zenEssentialTarget;
+        let switched;
+        if (isZenEssentialResult) {
+          switched = ChromeUtils.importESModule(
+            "resource:///modules/ZenUBEssentialsProvider.sys.mjs"
+          ).ZenUrlbarProviderEssentials.switchToTarget(
+            result.payload,
+            this.isPrivate
+          );
+        } else {
+          switched = this.window.switchToTabHavingURI(
+            Services.io.newURI(url),
+            true,
+            loadOpts,
+            lazy.UrlbarProviderOpenTabs.isNonPrivateUserContextId(
+              result.payload.userContextId
+            )
+              ? result.payload.userContextId
+              : null,
+            activeSplitView
+          );
+        }
         if (switched && !activeSplitView && prevTab.isEmpty) {
@@ -1785,3 +1797,3 @@ ${
         // properly unregistered when they are being closed.
-        if (!switched) {
+        if (!switched && !isZenEssentialResult) {
           console.error(`Tried to switch to non-existent tab: ${url}`);
