diff --git a/browser/components/urlbar/content/UrlbarView.mjs b/browser/components/urlbar/content/UrlbarView.mjs
index c0b378190..c7bcd39d4 100644
--- a/browser/components/urlbar/content/UrlbarView.mjs
+++ b/browser/components/urlbar/content/UrlbarView.mjs
@@ -2301,6 +2301,7 @@ export class UrlbarView {
     }
 
     item.removeAttribute("feedback-acknowledgment");
+    item.toggleAttribute("zen-essential", !!result.payload.zenEssentialTarget);
 
     if (
       result.type == UrlbarShared.RESULT_TYPE.SEARCH &&
